compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Markdown.hs

```diff
@@ -211,8 +211,11 @@ fct2md :: FunctionDoc -> RenderOut
 fct2md FunctionDoc{..} = mconcat
     [ renderAnchorInfix "" fct_anchor $ T.concat
         [ "**", escapeMd $ unFieldname fct_name, "**  " ]
-    , renderLinesDep $ \env ->
-        maybe [] (\t -> ["&nbsp; : " <> type2md env t]) fct_type
+    , renderLineDep $ \env -> T.concat
+        [ "&nbsp; : "
+        , maybe "" ((<> " => ") . type2md env) fct_context
+        , maybe "\\_"  (type2md env) fct_type
```

Comment:
When do we expect `fct_type` to be `Nothing`? For partial type signatures?

Reply:
Whenever we can't find the function's type info based on the name for some reason. Actually the only time this happens in the standard library is with the GHC.Enum functions (`eftInt`, `efdInt`, etc) and I'm not sure *why*.

Reply:
On further investigation, it seems this only happens when functions are not exported. (Does GHC change the names of non-exported functions, or get rid of the definitions altogether in some cases?) Since we shouldn't be showing these in the docs anyway, I think it'll be easy to make the type a mandatory part of the FunctionDocs.